### ISSUE TITLE
App Store Clones

### ISSUE THREAD

**Issue by DeepFriedPotato** [sprang/Inkpad#144]
Ink Pad Pro by Daniel Dugan
https://appsto.re/cn/WJib4.i

Ehh, he didn't even bother changing the icon = =

**Comment by 0101abc** [comment]
One more
https://itunes.apple.com/ru/app/!vectorpaint-vektornyj-redaktor/id789429915?mt=8

**Comment by 0101abc** [comment]
And this 
https://itunes.apple.com/ru/app/vector-paintpad-vektornyj/id989769728?mt=8

**Comment by ms-studio** [comment]
Is there a link to the original Inkpad app? is the app "inkPad 2" by Stone Studio Co Ltd a clone? The official Inkpad doesn't seem to exist in the Swiss app store...

**Comment by nainapjj** [comment]
Yes, same here.  I cannot seem to find a link to the open-source InkPad anywhere on the U.S. app store either.  Only these paid terrible clones.  Would it be okay if I submit a free version of Inkpad under my account? Is the reason Inkpad is gone is that the Taptrix developer account has expired?

**Comment by lagaisse** [comment]
I was looking for inkpad on the French AppStore but there is nothing except those terrible clones of clones.

**Comment by zeesi** [comment]
I seem to have been duped like others on this forum, I downloaded inkpad 2 by stone studio and paid not knowing it was a free app, thinking it was a new version, can't connect to dropbox, this is the only thing available in the app store, once I purchased there was a link to your site saying it is free, how can I complain and get my money refunded?

**Comment by JanX2** [comment]
@zeesi You have to contact Apple and request a refund.
http://www.imore.com/how-to-get-refund-itunes-app-store

**Comment by zeesi** [comment]
will this cause a corruption on my ipad? should I delete it ?
thank you for the quick response

thank you,
zeesi

**Comment by zeesi** [comment]
also where can I download the free original version of inkpad?

**Comment by Th3Ya0vi** [comment]
Is kind of funny when people don't know how to use open sources to create
there product.

**Comment by JanX2** [comment]
@zeesi This should not cause any changes apart from the Inkpad clone not working anymore. All I can currently find on the German app store are clones as well. The one I downloaded a year ago has disappeared.

**Comment by JanX2** [comment]
And the comments on the US app store for this Inkpad clone are… interesting:
https://itunes.apple.com/us/app/ink-pad-pro/id939821296?mt=8

Perhaps @sprang can help shed light onto all this. :)

**Comment by JanX2** [comment]
https://github.com/michaeldugan/Ink-Pad-Pro
“This repository is empty.”

**Comment by ghost** [comment]
Hi, I have a path to shut these down. Legal letter, apple contact, let's do it!

The main thing is that the names are not used, but even better is to get a better version out in the wild!

**Comment by Adrou** [comment]
Hi, I need a lot the original Inkpad app, I've restored my iPad and now I can't work anymore. I've found this: https://itunes.apple.com/us/app/inkpad-vector-design-illustration/id1057007769?mt=8

Anyone know if it is a clone? I'm at one step to purchase it :/

**Comment by alistairmcmillan** [comment]
@Adrou That is a clone. And it's one of the very few good clones. I'd definitely recommend it.

**Comment by Adrou** [comment]
@alistairmcmillan  thank you so much! Done, and you are right, I can work again.

**Comment by yury** [comment]
A little bit off topic, but why original app is not available in russian app store?

**Comment by JanX2** [comment]
@yury AFAIK it’s currently not available anywhere.

**Comment by ghost** [comment]
This one seems to be the only legit app I could find. https://itunes.apple.com/us/app/inkpad-vector-graphic-design/id1057007769?mt=8
It works great on IOS 10, haven't had any problems. I didn't mind paying since its the only app that is still being updated. I know not the original, but the original is dead.

**Comment by baburton** [comment]
Hum.. regarding the envoza fork (the Inkpad clone people are recommending above, which seems to still be on sale and under active development): I can't find where they are publishing their modifications to the MPL-licensed source code, and their website only links back to this 2-year-old repository.

I dearly hope I'm just missing something and they're not actively violating the license of the original authors whose software they are selling.

**Comment by alistairmcmillan** [comment]
@baburton For context, I'm one of the people who contributed to Inkpad while Steve was still working on this software.

A while ago I put together a list of everyone I could find who had an Inkpad clone in the store. I contacted everyone on the list and asked for their source code under the terms of the license (everyone I could find a way to contact anyway). Long story short, the couple behind the Envoza clone are the **only** ones who made their code available to me promptly and in a friendly manner.

If you want to see the code, please contact them. It's worth noting that the MPL doesn't require anyone to publish their source code automatically in public. It only requires that you provide the code on request to anyone who has received the executable. https://github.com/sprang/Inkpad/blob/develop/LICENSE#L174

**Comment by jackywongcw** [comment]
Hi @alistairmcmillan , I've gotten the executable and contacted them (the couple, Derek and Lindsay) about the source code, they seem to not respond to the email, I sent that 2 weeks ago. 
On the other hand, I've also sent you an email regarding this, and have not received any reply yet. 
Understand that you may be on a busy schedule and missed the email. So I'm posting here hopefully you could get a notification :)

**Comment by baburton** [comment]
Hi - so it's been a couple of years since this thread had any activity. The Envoza clone is not so attractive now: they require an ongoing paid subscription.

So: I've gone through and brought this up to date myself, put the source code on GitHub (https://github.com/baburton/Inkpad), and put it on the App Store for free: https://apps.apple.com/app/inkpad-libre/id1522113334 .

I plan to keep this app updated, because I use it also (indeed, I was also a user of Steve Sprang's version back in the day). Please let me know about any issues (or patches if you have them!) on the issue tracker at https://github.com/baburton/Inkpad .

- Ben.

**Comment by baburton** [comment]
> Hi @alistairmcmillan , I've gotten the executable and contacted them (the couple, Derek and Lindsay) about the source code, they seem to not respond to the email, I sent that 2 weeks ago.

For reference, I also have the executable and made a similar request for their modifications to the original sources. They explicitly said no. I do not know what their reasons are. - Ben.

**Comment by baburton** [comment]
> Hi @alistairmcmillan , I've gotten the executable and contacted them (the couple, Derek and Lindsay) about the source code, they seem to not respond to the email, I sent that 2 weeks ago.

@jackywongcw : FYI, I made a similar request (I had the executable and requested the modified sources), and they explicitly said no.

It would appear from a read of their "About" page that they have decided that the original authors' copyright and license are no longer relevant. (Their words: "Over the years ... the source code has been replaced".) I'm guessing their interpretation of a derived work is very different from mine, though of course IANAL, and I have no idea where their changes sit on the scale from (1) a line-by-line translation to different languages/frameworks versus (2) a full clean-room reimplementation. At any rate, I am also not one of the original copyright holders, so this probably makes my opinion irrelevant.

What I can suggest instead: if anyone is interested in working on the sources, I'd recommend starting from https://github.com/baburton/Inkpad, which is already up and running with iOS 13 and a modern iOS file browser interface, and importantly which remains free and open-source.

- Ben.
